octodns/record/svcb.py

```diff
@@ -0,0 +1,318 @@
+        ):
+            reasons.append(f'unsupported SvcParam "{mandatory}" in mandatory')
+        if mandatory.startswith('key'):
+            reasons += validate_svckey_number(mandatory)
+    return reasons
+
+
+def validate_svcparam_ech(svcparamvalue):
+    try:
+        b64decode(svcparamvalue, validate=True)
+    except binascii_error:
+        return ['ech SvcParam is invalid Base64']
+
+
+def validate_svckey_number(paramkey):
+    try:
+        paramkeynum = int(paramkey[3:])
+        if 7 < paramkeynum > 65535:
+            return [f'SvcParam key "{paramkey}" has wrong key number']
+    except ValueError:
+        return [f'SvcParam key "{paramkey}" has wrong format']
+    return []
+
+
+def parse_rdata_text_svcparamvalue_list(svcparamvalue):
+    return svcparamvalue.split(',')
+
+
+def svcparamkeysort(svcparamkey):
+    if svcparamkey.startswith('key'):
+        return int(svcparamkey[3:])
+    return SUPPORTED_PARAMS[svcparamkey]['key_num']
+
+
+# cc https://datatracker.ietf.org/doc/html/rfc9460#keys
+SUPPORTED_PARAMS = {
+    'no-default-alpn': {'key_num': 2, 'has_arg': False},
+    'alpn': {
+        'key_num': 1,
+        'validate': validate_svcparam_alpn,
+        'parse_rdata_text': parse_rdata_text_svcparamvalue_list,
+    },
+    'port': {'key_num': 3, 'validate': validate_svcparam_port},
+    'ipv4hint': {
+        'key_num': 4,
+        'validate': validate_svcparam_ipv4hint,
+        'parse_rdata_text': parse_rdata_text_svcparamvalue_list,
+    },
+    'ipv6hint': {
+        'key_num': 6,
+        'validate': validate_svcparam_ipv6hint,
+        'parse_rdata_text': parse_rdata_text_svcparamvalue_list,
+    },
+    'mandatory': {
+        'key_num': 0,
+        'validate': validate_svcparam_mandatory,
+        'parse_rdata_text': parse_rdata_text_svcparamvalue_list,
+    },
+    'ech': {'key_num': 5, 'validate': validate_svcparam_ech},
+}
+
+
+class SvcbValue(EqualityTupleMixin, dict):
+
+    @classmethod
+    def parse_rdata_text(cls, value):
+        try:
+            (svcpriority, targetname, *svcparams) = value.split(' ')
+        except ValueError:
+            raise RrParseError()
+        try:
+            svcpriority = int(svcpriority)
+        except ValueError:
+            pass
+        targetname = unquote(targetname)
+        params = dict()
+        for svcparam in svcparams:
+            paramkey, *paramvalue = svcparam.split('=')
+            if paramkey in params.keys():
+                raise RrParseError(f'{paramkey} is specified twice')
+            if len(paramvalue) != 0:
+                params[paramkey] = paramvalue[0]
+                if (
+                    SUPPORTED_PARAMS.get(paramkey, {}).get(
+                        'parse_rdata_text', None
+                    )
+                    is not None
+                ):
+                    params[paramkey] = SUPPORTED_PARAMS[paramkey][
+                        'parse_rdata_text'
+                    ](paramvalue[0])
```

Comment:
<img width="1804" alt="Screenshot 2024-06-05 at 2 33 12 PM" src="https://github.com/octodns/octodns/assets/12789/5c063a6a-3494-41e8-846d-e400967111ba">

I think the failures have to do with a ~bug in older versions of python <-> coverage. I vagulely remembered running into something similar in the past and github's `involves:ross` search in the coverage repo turned it up:

https://github.com/nedbat/coveragepy/issues/1672

I believe the following will keep the same logic/intent, but hopefully break things up in a way that avoids the optimization/coverage problem. It should also avoid double looking things up in the dicts. 

```suggestion
                parse_rdata_text = SUPPORTED_PARAMS.get(paramkey, {}).get(
                    'parse_rdata_text', None
                )
                if parse_rdata_text is not None:
                    params[paramkey] = parse_rdata_text
```

Reply:
Fixed!